src/aead/aead.rs

```diff
@@ -613,4 +633,46 @@ mod tests {
                                         &mut in_out, &ad).is_err());
         }
     }
+
+    fn test_aead_inner(aead_alg: &'static aead::Algorithm,
+                       key_bytes: &[u8],
+                       nonce: &[u8],
+                       plaintext: &Vec<u8>,
+                       ad: &[u8],
+                       ct: &[u8],
+                       error: &Option<String>,
+                       in_prefix: &[u8]) {
+        let max_overhead_len = aead_alg.max_overhead_len();
+        let mut s_in_out = plaintext.clone();
+        for _ in 0..max_overhead_len {
+            s_in_out.push(0);
+        }
+        let s_key = aead::SealingKey::new(aead_alg, key_bytes).unwrap();
+        let s_result = aead::seal_in_place(&s_key, nonce,
+                                           &mut s_in_out[..],
+                                           max_overhead_len, &ad);
+
+        let mut o_in_out = Vec::from(in_prefix);
+        o_in_out.extend_from_slice(&ct[..]);
```

Comment:
Here's an idea:

1. Construct a Vec with `4096 + ct.len()` capacity.
2. For each `in_prefix_len`, reset the vec to zero length, then append `in_prefix_len` bytes to it, then append `ct` to it.

This way, only one memory allocation will be done per test vector, instead of one memory allocation per in_prefix_len per test vector.